### ISSUE TITLE
How can I save the last-modified field of the image for next use

### ISSUE THREAD

**Issue by wickey-w** [SDWebImage/SDWebImage#2880]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dreampiggy** [comment]
Did you means HTTP response header ?https://developer.mozilla.org/docs/Web/HTTP/Headers/Last-Modified

You can use the URL Respose Modifier. But this is the 5.3.0 features. See: https://github.com/SDWebImage/SDWebImage/blob/master/SDWebImage/Core/SDWebImageDownloaderResponseModifier.h

For 5.0-5.2, you can just check the `NSURLHTTPResponse` by yourself, but not override. Or maybe you can using [Custom Download](https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#custom-download-operation-40) to override https://developer.apple.com/documentation/foundation/nsurlsessiondatadelegate/1411612-urlsession?language=objc

```objective-c
SDWebImageDownloadToken *token = [SDWebImageDownloader downloadImageWithURL:url];
NSURLHTTPResponse *response = (NSURLHTTPResponse *)token.operation.response;
NSDictionary *allHeaders = response.allHeaderFilelds;
```

**Comment by wickey-w** [comment]
我用的是5.1的版本，我想问怎么自定义下载，获取响应头保存，有demo示例吗？

**Comment by dreampiggy** [comment]
Write the demo code just from logic, you can have a test on simulator or device.

```objective-c
@interface MyDownloaderOperation: SDWebImageDownloaderOperation<SDWebImageDownloaderOperation>

@end

@implementation MyDownloaderOperation

- (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)dataTask willCacheResponse:(NSCachedURLResponse *)proposedResponse completionHandler:(void (^)(NSCachedURLResponse *cachedResponse))completionHandler {
    NSHTTPURLResponse *response = (NSHTTPURLResponse *)proposedResponse.response;
    NSMutableDictionary *modifiedHeaders = [response.allHeaderFields mutableCopy];
    modifiedHeaders[@"Last-Modified"] = @"Wed, 21 Oct 2015 07:28:00 GMT";
    NSHTTPURLResponse *modifiedResponse = [[NSHTTPURLResponse alloc] initWithURL:response.URL statusCode:response.statusCode HTTPVersion:nil headerFields:modifiedHeaders.copy];
    NSCachedURLResponse *cachedResponse = [[NSCachedURLResponse alloc] initWithResponse:modifiedResponse data:proposedResponse.data];
    completionHandler(cachedResponse);
}

// Then register the custom downloader operation class in SDWebImageDownloaderConfig.operationClass;
@end
```

**Comment by wickey-w** [comment]
你这种方法是获取response的内容，怎么保存到本地，供下次使用呢？

**Comment by dreampiggy** [comment]
@wickey-w URLSession will call the `willCacheResponse` method when it need to save the cached response. This is default behavior for `NSURLRequestUseProtocolCachePolicy`. (For SDWebImage, you can use `refreshCached` options, or custom request modifier)

And, [5.3.0](https://github.com/SDWebImage/SDWebImage/releases/tag/5.3.0) is released. Checkout the release changelog here and [wiki](https://github.com/SDWebImage/SDWebImage/wiki/How-to-use#use-response-modifier-530) about response modifier.
